src/test/java/net/floodlightcontroller/statistics/GetCurrentPortSpeedTest.java

```diff
@@ -0,0 +1,145 @@
+package net.floodlightcontroller.statistics;
+
+import net.floodlightcontroller.core.FloodlightContext;
+import net.floodlightcontroller.core.IOFSwitch;
+import net.floodlightcontroller.core.internal.IOFSwitchService;
+import net.floodlightcontroller.core.module.FloodlightModuleContext;
+import net.floodlightcontroller.core.test.MockThreadPoolService;
+import net.floodlightcontroller.core.types.NodePortTuple;
+import net.floodlightcontroller.test.FloodlightTestCase;
+import org.easymock.EasyMock;
+import org.projectfloodlight.openflow.protocol.*;
+
+import org.projectfloodlight.openflow.types.DatapathId;
+import org.projectfloodlight.openflow.types.OFPort;
+import org.projectfloodlight.openflow.protocol.OFFactories;
+import net.floodlightcontroller.threadpool.IThreadPoolService;
+
+import java.util.*;
+import static org.easymock.EasyMock.*;
+import static org.junit.Assert.*;
+import org.junit.*;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+/**
+ * Created by qing wang on 5/18/17.
+ */
+@RunWith(Parameterized.class)
+public class GetCurrentPortSpeedTest extends FloodlightTestCase {
+
+    private FloodlightContext cntx;
+    private FloodlightModuleContext fmc;
+    private MockThreadPoolService threadpool;
+    private IOFSwitchService switchService;
+    private StatisticsCollector statsCollector;
+
+    private static final OFFactory factory11 = OFFactories.getFactory(OFVersion.OF_11);
+    private static final OFFactory factory12 = OFFactories.getFactory(OFVersion.OF_12);
+    private static final OFFactory factory13 = OFFactories.getFactory(OFVersion.OF_13);
+    private static final OFFactory factory14 = OFFactories.getFactory(OFVersion.OF_14);
+    private static final OFFactory factory15 = OFFactories.getFactory(OFVersion.OF_15);
+    private static OFFactory inputFactory;
+    private static Long expectedSpeed;
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+
+        cntx = new FloodlightContext();
+        fmc = new FloodlightModuleContext();
+
+        // Module loader setup
+        threadpool = new MockThreadPoolService();
+        switchService = getMockSwitchService();
+        statsCollector = new StatisticsCollector();
+
+        fmc.addService(IThreadPoolService.class, threadpool);
+        fmc.addService(IOFSwitchService.class, switchService);
+
+        threadpool.init(fmc);
+        statsCollector.init(fmc);
+        threadpool.startUp(fmc);
+
+    }
+
+    /**
+     * This constructor will be called for each row of test data collection
+     * @param inputFactory
+     * @param expectedSpeed
+     */
+    public GetCurrentPortSpeedTest(OFFactory inputFactory, Long expectedSpeed) {
+        this.inputFactory = inputFactory;
+        this.expectedSpeed = expectedSpeed;
+    }
+
+    /**
+     * A Collection of Junit Test with various "inputFactory" and "expectedSpeed"
+     * @return
+     */
```

Comment:
This is right. The way to use parameterized Junit test is to create this Collection object as  "test Data Set or parameter set". Then we need to use a public constructor to receive one row of that test data once a time. Because of using this public constructor, I then consider to create a new class to just test "current port speed". But you are right, I could use different constructor to receive different test data, such that we can have one single statistic class with multiple tests.